nifi-nar-bundles/nifi-dropbox-bundle/nifi-dropbox-processors/src/main/java/org/apache/nifi/processors/dropbox/PutDropbox.java

```diff
@@ -0,0 +1,324 @@
+    ));
+
+    private static final Set<Relationship> RELATIONSHIPS;
+
+    static {
+        final Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_FAILURE);
+        RELATIONSHIPS = Collections.unmodifiableSet(rels);
+    }
+
+    private DbxClientV2 dropboxApiClient;
+
+    private DbxUploader<?, ?, ?> dbxUploader;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        dropboxApiClient = getDropboxApiClient(context, getIdentifier());
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String folder = context.getProperty(FOLDER).evaluateAttributeExpressions(flowFile).getValue();
+        final String filename = context.getProperty(FILE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+
+        final long chunkUploadThreshold = context.getProperty(CHUNKED_UPLOAD_THRESHOLD)
+                .asDataSize(DataUnit.B)
+                .longValue();
+
+        final long uploadChunkSize = context.getProperty(CHUNKED_UPLOAD_SIZE)
+                .asDataSize(DataUnit.B)
+                .longValue();
+
+        final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+
+        boolean uploadErrorOccurred = false;
+
+        final long size = flowFile.getSize();
+        final String uploadPath = convertFolderName(folder) + "/" + filename;
+
+        try (final InputStream rawIn = session.read(flowFile)) {
+            try {
+                if (size <= chunkUploadThreshold) {
+                    try (UploadUploader uploader = createUploadUploader(uploadPath, conflictResolution)) {
+                        uploader.uploadAndFinish(rawIn);
+                    }
+                } else {
+                    uploadLargeFileInChunks(uploadPath, rawIn, size, uploadChunkSize, conflictResolution);
+                }
+            } catch (UploadErrorException e) {
+                handleUploadError(conflictResolution, uploadPath, e);
+            } catch (RateLimitException e) {
+                context.yield();
+                throw new ProcessException("Dropbox API rate limit exceeded while uploading file", e);
+            }
+        } catch (Exception e) {
+            getLogger().error("Exception occurred while uploading file '{}' to Dropbox folder '{}'", filename, folder, e);
+            uploadErrorOccurred = true;
+        } finally {
+            dbxUploader.close();
+        }
+
+        if (!uploadErrorOccurred) {
+            session.transfer(flowFile, REL_SUCCESS);
+        } else {
+            session.transfer(flowFile, REL_FAILURE);
+        }
```

Comment:
I would be a clearer code design without the `uploadErrorOccurred` flag. The success case can be moved into the `try` block and the failure into `catch`.

Please also send a provenance event for success and penalize the FlowFile in case of failure.
Example can be found in [PutAzureDataLakeStorage](https://github.com/apache/nifi/blob/ed6ba537249ea781402dc0ac7e341907ba5b0c94/nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureDataLakeStorage.java#L144-L157).
Please note, provenance event is only needed when the upload has really happened (so not when the file exists and conflict strategy is ignore).